## Day 1 — Descriptive Statistics

### task1: Load a numeric column from a provided dataset into a NumPy array or Pandas Series.

In [2]:
import pandas as pd
import numpy as np
df = pd.read_csv('Mobile Price/train.csv')
df.columns

Index(['battery_power', 'blue', 'clock_speed', 'dual_sim', 'fc', 'four_g',
       'int_memory', 'm_dep', 'mobile_wt', 'n_cores', 'pc', 'px_height',
       'px_width', 'ram', 'sc_h', 'sc_w', 'talk_time', 'three_g',
       'touch_screen', 'wifi', 'price_range'],
      dtype='str')

In [ ]:
battery_power = df['battery_power']
battery_power # type: pandas.core.series.Series

0        842.0
1       1021.0
2        563.0
3          NaN
4       1821.0
         ...  
1995     794.0
1996    1965.0
1997    1911.0
1998    1512.0
1999     510.0
Name: battery_power, Length: 2000, dtype: float64

### task2: Compute its mean, median, mode, standard deviation, and IQR.

In [6]:
mean_battery_power = battery_power.mean()
median_battery_power = battery_power.median()
mode_battery_power = battery_power.mode()[0]  # mode() returns a Series, take the first value
std_battery_power = battery_power.std()
IQR_battery_power = battery_power.quantile(0.75) - battery_power.quantile(0.25)
print(f"Mean: {mean_battery_power}")
print(f"Median: {median_battery_power}")
print(f"Mode: {mode_battery_power}")
print(f"Standard Deviation: {std_battery_power}")
print(f"Interquartile Range (IQR): {IQR_battery_power}")

Mean: 1239.064064064064
Median: 1228.0
Mode: 618.0
Standard Deviation: 439.2923600135214
Interquartile Range (IQR): 763.75


Statistical analysis of the battery_power column shows a mean of 1239.06 and a median of 1228.0, and their closeness indicates the data is roughly symmetric, without strong skewness or influential outliers. The mode came out at 618.0, the most frequently occurring value, despite its relative distance from the center — which is normal for continuous, varied data like battery capacity. The standard deviation was 439.29, roughly 35% of the mean, reflecting moderate-to-high spread across devices. The interquartile range (IQR) was 763.75, a large value consistent with the standard deviation, confirming that the spread comes from genuine variation in battery specifications across devices rather than from outliers requiring cleaning. Overall, the column represents a healthy, well-behaved distribution suitable for use as a feature in a subsequent classification model.

### task3: Identify whether the mean or median better represents a 'typical' value, and justify the choice.

The mean and median for battery_power are nearly identical (1239.06 vs. 1228.0, a difference of less than 1%), which indicates the distribution is roughly symmetric with no significant skew or influential outliers. Because of this, the mean is the better representative of a "typical" value: it uses all the data points rather than just the middle position, and it integrates naturally with other statistics like standard deviation for any downstream analysis. The usual reason to prefer the median — protection against distortion by extreme values — doesn't apply here, since the IQR (763.75) is consistent with the standard deviation (439.29), confirming there's no meaningful outlier problem. Had the mean and median diverged substantially, that would have signaled skew, and the median would have become the safer choice instead.

### task4:  Write a Markdown cell summarizing the column's center and spread in plain language.

The `battery_power` column has an average value of about **1239 mAh**, with the median 
sitting close by at **1228 mAh**. Since these two values are so close together, the data 
appears fairly balanced — there's no strong skew pulling it in one direction, and no 
extreme values distorting the average.

On the other hand, the values are fairly spread out: the standard deviation is about 
**439 mAh**, and the middle 50% of devices (IQR) span a wide range of about **764 mAh**. 
This tells us that phone battery capacities in this dataset vary quite a bit — from lower-
capacity devices to much higher-capacity ones — which is expected and reflects real 
diversity in the phones being compared, rather than errors or outliers in the data.